src/CADL.Extension/Emitter.Csharp/src/lib/utils.ts

```diff
@@ -1,3 +1,16 @@
+import { Model } from "@typespec/compiler";
+
 export function capitalize(str: string): string {
     return str.charAt(0).toUpperCase() + str.slice(1);
 }
+
+export function getNameForTemplate(model: Model): string {
```

Comment:
Can we add unit test for this?  I believe this part is easy to verify in emitter

Reply:
Added a follow up since this is high priority change if you can add this in when you are back next week https://github.com/Azure/autorest.csharp/issues/3395.